plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -18,38 +18,96 @@ const updateReviewsStatuses: ActionPluginType = async ({
-      [ReviewStatus.Submitted, ReviewStatus.Locked, ReviewStatus.Draft].includes(
-        review.reviewStatus
-      )
-    )
-    // Deduce which ones should be updated
-    for (const review of reviews) {
-      const { reviewAssignmentId, levelNumber, reviewStatus } = review
-      if (levelNumber > 1) reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (reviewStatus === ReviewStatus.Locked)
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
+    if (isReview) {
+      // Review submitted from upper level to lower level review
+      if (decision === Decision.ChangesRequested) {
+        const previousLevelReview = currentReviewLevel - 1
+        // Update lower level reviews with assigned responses on the array changed responses ...
+        const reviewsSubmitted = await getReviewsByLevelAndStatus(previousLevelReview, [
+          ReviewStatus.Submitted,
+        ])
+        // Deduce which ones should be updated to Pending
+        for (const review of reviewsSubmitted) {
+          const { reviewAssignmentId } = review
+          if (await haveAssignedResponsesChanged(reviewAssignmentId))
+            reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.ChangesRequested })
+        }
+      } else {
+        // Review submitted from lower level to upper level
+        // Update same level reviews as submitted if no other still have changes requested status...
+        const othersChangeRequested = await getReviewsByLevelAndStatus(currentReviewLevel, [
+          ReviewStatus.ChangesRequested,
+        ])
+        if (othersChangeRequested.length === 0) {
+          // Get all Locked reviews matching application_id, current stage, current level and in Locked status
+          const reviewsLocked = await getReviewsByLevelAndStatus(currentReviewLevel, [
+            ReviewStatus.Locked,
+          ])
+
+          // Now previous locked  to be allowed to continue
+          // Locked -> to avoid other reviews submitted while awaiting changes requests
+          reviewsLocked.forEach((review) =>
+            reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Draft })
+          )
+        }
+
+        // Update upper level reviews submitted
+        const nextReviewLevel = currentReviewLevel + 1
+        const submittedReviews = await getReviewsByLevelAndStatus(nextReviewLevel, [
+          ReviewStatus.Submitted,
+        ])
+        submittedReviews.forEach((review) =>
+          reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
+        )
+      }
+    } else {
+      // For Application submission
+      // Get reviews/review assignments (with status) matching application_id & current stage
+      const reviews = await getReviewsByLevelAndStatus(currentReviewLevel || 1, [
```

Comment:
maybe we should always just use level 1 when application submitted

Reply:
Yes, I think that would be ok

Reply:
Done: bddc898